### ISSUE TITLE
Compiler-Error during Code-Compilation for Due

### ISSUE THREAD

**Issue by bfarwick** [ZinggJM/ILI9486_SPI#1]
Hi,
I tried to compile a short test-program to test a Waveshare 4inch TFT Touch Shield Display on a Arduino Due.
IDE: Arduino 1.8.12
Code like this:
#include <SPI.h>
#include "ILI9486_SPI.h"

ILI9486_SPI tft(/*CS=10*/ 10, /*DC=*/ 7, /*RST=*/ 8);

#define  BLACK   0x0000
#define BLUE    0x001F
#define RED     0xF800
#define GREEN   0x07E0
#define CYAN    0x07FF
#define MAGENTA 0xF81F
#define YELLOW  0xFFE0
#define WHITE   0xFFFF

void setup() {
  // put your setup code here, to run once:
  tft.init();
  tft.fillScreen(RED);
}

void loop() {
  // put your main code here, to run repeatedly:
  for (uint8_t r = 0; r < 4; r++)
  {
    //Serial.print("rotation "); Serial.println(r);
    tft.setRotation(r);
    tft.fillScreen(BLACK);
    tft.setCursor(0, 0);
    tft.setTextColor(RED, BLACK);
    tft.setTextSize(1);
    tft.println("Hello World!");
    tft.println(01234.56789, 2);
    tft.println(0xDEADBEF, HEX);

    tft.setTextColor(GREEN);
    tft.setTextSize(2);
    tft.println("Hello");
    tft.println(01234.56789, 2);
    tft.println(0xDEADBEF, HEX);

    tft.setTextColor(BLUE);
    tft.setTextSize(3);
    tft.println("Hello");
    tft.println(01234.56789, 2);
    tft.println(0xDEADBEF, HEX);
    //Serial.print("rotation "); Serial.print(r); Serial.println(" done");

    delay(5000);
  }
}

Error-Message:
Arduino: 1.8.12 (Windows 10), Board: "Arduino Due (Programming Port)"

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp: In constructor 'ILI9486_SPI::ILI9486_SPI(int8_t, int8_t, int8_t)':

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:44:9: error: assignment of read-only member 'Adafruit_GFX::WIDTH'

   WIDTH = 320;

         ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:45:10: error: assignment of read-only member 'Adafruit_GFX::HEIGHT'

   HEIGHT = 480;

          ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp: In member function 'void ILI9486_SPI::_setWindow(uint16_t, uint16_t, uint16_t, uint16_t)':

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:409:29: warning: narrowing conversion of '(((int)x) >> 8)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t columns[] = {x >> 8, x & 0xFF, xe >> 8, xe & 0xFF};

                             ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:409:37: warning: narrowing conversion of '(((int)x) & 255)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t columns[] = {x >> 8, x & 0xFF, xe >> 8, xe & 0xFF};

                                     ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:409:48: warning: narrowing conversion of '(((int)xe) >> 8)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t columns[] = {x >> 8, x & 0xFF, xe >> 8, xe & 0xFF};

                                                ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:409:57: warning: narrowing conversion of '(((int)xe) & 255)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t columns[] = {x >> 8, x & 0xFF, xe >> 8, xe & 0xFF};

                                                         ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:412:26: warning: narrowing conversion of '(((int)y) >> 8)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t rows[] = {y >> 8, y & 0xFF, ye >> 8, ye & 0xFF};

                          ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:412:34: warning: narrowing conversion of '(((int)y) & 255)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t rows[] = {y >> 8, y & 0xFF, ye >> 8, ye & 0xFF};

                                  ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:412:45: warning: narrowing conversion of '(((int)ye) >> 8)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t rows[] = {y >> 8, y & 0xFF, ye >> 8, ye & 0xFF};

                                             ^

C:\Users\Bernie\Documents\Arduino\libraries\ILI9486_SPI-master\src\ILI9486_SPI.cpp:412:54: warning: narrowing conversion of '(((int)ye) & 255)' from 'int' to 'uint16_t {aka short unsigned int}' inside { } [-Wnarrowing]

     uint16_t rows[] = {y >> 8, y & 0xFF, ye >> 8, ye & 0xFF};

                                                      ^

exit status 1
Fehler beim Kompilieren für das Board Arduino Due (Programming Port).

Any Idea, what's wrong? Thank's for help, Bernie

**Comment by ZinggJM** [comment]
@bfarwick,

thank you for reporting this issue.

I confirm this is an error, a leftover from the copy from GxTFT.
But it compiles ok also for Arduino Due with the actual version of Adafruit_GFX.
The protected members are not const:
```
  int16_t WIDTH,      ///< This is the 'raw' display width - never changes
      HEIGHT;         ///< This is the 'raw' display height - never changes
```

Do you use a different version?

I will fix this, sometime soon.

**Comment by ZinggJM** [comment]
fixed in Version 1.0.1
